politeiawww/politeiad.go

```diff
@@ -258,8 +400,120 @@ func (p *politeiawww) getVetted(token, version string) (*pd.Record, error) {
 	return &gvr.Record, nil
 }
 
-// getVettedLatest returns the latest version of the vvetted record for the
+// getVettedLatest returns the latest version of the vetted record for the
 // provided token.
 func (p *politeiawww) getVettedLatest(token string) (*pd.Record, error) {
 	return p.getVetted(token, "")
 }
+
+// pluginInventory requests the plugin inventory from politeiad and returns
+// inventoryByStatus retrieves the censorship record tokens filtered by status.
+func (p *politeiawww) inventoryByStatus() (pd.InventoryByStatusReply, error) {
+	// Setup request
+	challenge, err := util.Random(pd.ChallengeSize)
+	if err != nil {
+		return pd.InventoryByStatusReply{}, err
+	}
+	ibs := pd.InventoryByStatus{
+		Challenge: hex.EncodeToString(challenge),
+	}
+
+	// Send request
+	resBody, err := p.makeRequest(http.MethodPost, pd.InventoryByStatusRoute, ibs)
+	if err != nil {
+		return pd.InventoryByStatusReply{}, err
+	}
+
+	// Receive reply
+	var ibsr pd.InventoryByStatusReply
+	err = json.Unmarshal(resBody, &ibsr)
+	if err != nil {
+		return pd.InventoryByStatusReply{}, err
+	}
+
+	// Verify challenge
+	err = util.VerifyChallenge(p.cfg.Identity, challenge, ibsr.Response)
+	if err != nil {
+		return pd.InventoryByStatusReply{}, err
+	}
+
+	return ibsr, nil
+}
+
+// pluginInventory requests the plugin inventory from politeiad and returns
+// the available plugins slice.
+func (p *politeiawww) pluginInventory() ([]Plugin, error) {
```

Comment:
This should be returning the `pd.Plugin` type.  Its up to the calling function to decide what to do with it.